examples/IB/explicit/ex7/example.cpp

```diff
@@ -0,0 +1,515 @@
+
+
+            dt = time_integrator->getMaximumTimeStepSize();
+            time_integrator->advanceHierarchy(dt);
+            loop_time += dt;
+
+            pout << "\n";
+            pout << "At end       of timestep # " << iteration_num << "\n";
+            pout << "Simulation time is " << loop_time << "\n";
+            pout << "+++++++++++++++++++++++++++++++++++++++++++++++++++\n";
+            pout << "\n";
+
+            //Print the Eulerian Coordinates of the IB points which constitute the Lagrangian structure.
+            const int finest_ln = patch_hierarchy->getFinestLevelNumber();
+            Pointer<LData> X_data = ib_method_ops->getLDataManager()->getLData("X", finest_ln);
+            Vec X_vec = X_data->getVec();
+            double* x_vals;
+            int ierr = VecGetArray(X_vec, &x_vals);
+            IBTK_CHKERRQ(ierr);
+            Pointer<LMesh> l_mesh = ib_method_ops->getLDataManager()->getLMesh(finest_ln);
+            const std::vector<LNode*>& local_nodes = l_mesh->getLocalNodes();
+            int num_nodes = local_nodes.size();
+            //Send to processor zero the number of IB points on each processor
+            if(rank == 0)
+            {
+                nodes_per_proc[0] = num_nodes;
+            }
+
+            if(rank !=0)
+            {
+                MPI_Send(&num_nodes,1,MPI_INT,0,0,IBTK_MPI::getCommunicator());
+            }
+            else
+            {
+                for(int r = 1; r<n_proc; ++r)
+                {
+                    MPI_Recv(&nodes_per_proc[r],1,MPI_INT,r,0,IBTK_MPI::getCommunicator(),MPI_STATUS_IGNORE);
+                }
+            }
+
+            IBTK_MPI::barrier();
+            //initialize vector to store the nodal coordinates on each processor
+            std::vector<double> Local_Coords(num_nodes*NDIM);
+            int k = 0;
+
+                for (const auto& node : local_nodes)
+                {
+
+
+                    const int petsc_idx = node->getLocalPETScIndex();
+                    Eigen::Map<Vector3d> X(&x_vals[petsc_idx * NDIM]); // X is a vector containing the local coordinates
+                    for(int m = 0; m<NDIM;m++)
+                    {
+                        Local_Coords[NDIM*k + m] = X[m];
+                    }
+
+                    k = k+1;
+
+                }
+            //Compute a vector containing the local node coordinates on each processor to be sent via MPI_Gatherv
+
+            IBTK_MPI::barrier();
+            //Initialize a vector to recieve the coordinates to print to
+            std::vector<double> RECV_BUF;
+            //Specify what's done on the root process
+            int total_nodes = 0;
+            if(rank == 0)
+            {
+
+                for(std::vector<int>::iterator it = nodes_per_proc.begin(); it != nodes_per_proc.end(); ++it)
+                     {
+                        total_nodes += *it;
+                     }
```

Comment:
```suggestion
                const auto total_nodes = std::accumulate(nodes_per_proc.begin(), nodes_per_proc.end(), 0);
```
same thing but shorter. Like elsewhere, this should be 'global' if it refers to everything (not total).